google-cloud-tools-plugin/testSrc/com/google/gct/idea/debugger/ui/CloudAttachDialogTest.java

```diff
@@ -0,0 +1,156 @@
+package com.google.gct.idea.debugger.ui;
+
+import static org.mockito.Mockito.when;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.services.clouddebugger.model.Debuggee;
+import com.google.gct.idea.elysium.ProjectSelector;
+import com.google.gct.idea.util.GctBundle;
+import com.google.gct.login.CredentialedUser;
+import com.google.gct.login.GoogleLogin;
+import com.google.gct.login.MockGoogleLogin;
+import com.google.gdt.eclipse.login.common.GoogleLoginState;
+
+import com.intellij.openapi.ui.ValidationInfo;
+import com.intellij.testFramework.PlatformTestCase;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.LinkedHashMap;
+
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+
+public class CloudAttachDialogTest extends PlatformTestCase {
+  private static final String USER = "test@user.com";
+  private static final String PASSWORD = "123";
+  private CredentialedUser user;
+
+  private ProjectSelector projectSelector;
+  private CloudAttachDialog dialog;
+  private JComboBox moduleSelector;
+  private JLabel warningLabelText;
+  private JLabel warningLabelHeader;
+
+  @Before
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    mockCredentials();
+
+    dialog = new CloudAttachDialog(this.getProject());
+    projectSelector = dialog.getMyElysiumProjectId();
+    moduleSelector = dialog.getMyDebuggeeTarget();
+    warningLabelText = dialog.getMyWarningLabel2();
+    warningLabelHeader = dialog.getMyWarningLabel();
+  }
+
+  @Test
+  public void testErrorWhenUserIsLoggedOut() {
+    mockLoggedOutUser();
+    ValidationInfo error = dialog.doValidate();
+
+    assertNotNull(error);
+    assertEquals(GctBundle.getString("clouddebug.nologin"), error.message);
+  }
+
+  @Test
+  public void testErrorWhenNoProjectSelected() {
+    mockLoggedInUser();
+    ValidationInfo error = getValidationError();
+
+    assertNotNull(error);
+    assertEquals(GctBundle.getString("clouddebug.noprojectid"), error.message);
+  }
+
+  @Test
+  public void testNoModulesFound() {
+    mockLoggedInUser();
+    selectEmptyProject();
+    ValidationInfo error = getValidationError();
+
+    // Errors
+    assertNotNull(error);
+    assertEquals(GctBundle.getString("clouddebug.selectvalidproject"), error.message);
```

Comment:
I prefer not to use same constant in tests and model code. Use the literal here and then we notice if the constant is not what it should be.